src/plugins/topic_viewer/TopicViewer.hh

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+#include <QStandardItem>
+#include <QStandardItemModel>
+#include <QHash>
+#include <QByteArray>
+#include <QString>
+#include <QModelIndex>
+#include <string>
+#include <vector>
+#include <deque>
+#include <memory>
+
+#include <ignition/gui/Application.hh>
+#include <ignition/gui/Plugin.hh>
+
+#include <ignition/msgs.hh>
+#include <ignition/transport.hh>
+#include <ignition/transport/Node.hh>
+#include <ignition/transport/MessageInfo.hh>
+#include <ignition/transport/Publisher.hh>
+#include <ignition/common.hh>
+#include <ignition/gui.hh>
+
+namespace ignition
+{
+namespace gui
+{
+namespace plugins
+{
+
+  class TopicsModel;
+  class TopicViewerPrivate;
+
+  /// \brief Model for the Topics and their Msgs and Fields
+  /// a tree model that represents the topics tree with its Msgs
+  /// Childeren and each msg node has its own fileds/msgs childeren
+  class TopicViewer : public Plugin
+  {
+    Q_OBJECT
+
+    /// \brief Constructor
+    public: TopicViewer();
+
+    /// \brief Destructor
+    public: ~TopicViewer();
+
+    /// \brief Documentaation inherited
+    public: void LoadConfig(const tinyxml2::XMLElement *) override;
+
+    /// \brief Create the fields model
+    private: void CreateModel();
+
+    /// \brief add a topic to the model
+    /// \param[in] _topic topic name to be displayed
+    /// \param[in] _msg topic's msg type
+    private: void AddTopic(const std::string &_topic,
+                           const std::string &_msg);
+
+    /// \brief add a field/msg child to that parent item
+    /// \param[in] _parentItem a parent for the added field/msg
+    /// \param[in] _msgName the displayed name of the field/msg
+    /// \param[in] _msgType field/msg type
+    private: void AddField(QStandardItem *_parentItem,
+                           const std::string &_msgName,
+                           const std::string &_msgType);
+
+    /// \brief factory method for creating an item
+    /// \param[in] _name the display name
+    /// \param[in] _type type of the field of the item
+    /// \param[in] _path a set of concatinate strings of parent msgs
```

Comment:
```suggestion
    /// \param[in] _path a set of concatenate strings of parent msgs
```

It would be helpful to add an example here on the documentation